src/components/ExportButton.tsx

```diff
@@ -0,0 +1,354 @@
+
+  /** Stop any ongoing recordings and reset the current frame, optionally closing the modal. */
+  const stopRecording = useCallback(
+    (closeModal: boolean) => {
+      props.stopRecording();
+      // Reset the frame number (clean up!)
+      props.setFrame(originalFrame);
+      setIsRecording(false);
+      setPercentComplete(0);
+      if (closeModal) {
+        setIsLoadModalOpen(false);
+      }
+    },
+    [originalFrame, props.stopRecording]
+  );
+
+  /**
+   * Triggered when the user attempts to cancel or exit the main modal.
+   */
+  const handleCancel = useCallback(() => {
+    // Not recording; exit
+    if (!isRecording) {
+      setIsLoadModalOpen(false);
+      return;
+    }
+
+    // Currently recording; user must be prompted to confirm
+    modal.confirm({
+      title: "Cancel export",
+      content: "Are you sure you want to cancel and exit?",
+      okText: "Cancel",
+      cancelText: "Back",
+      centered: true,
+      icon: null,
+      getContainer: modalContextRef.current || undefined,
+      onOk: () => {
+        stopRecording(true);
+      },
+    });
+  }, [isRecording, modalContextRef.current, stopRecording]);
+
+  /**
+   * Stop the recording without closing the modal.
+   */
+  const handleStop = useCallback(() => {
+    stopRecording(false);
+  }, [stopRecording]);
+
+  // Note: This is not wrapped in a useCallback because it has a large number
+  // of dependencies, and is likely to update whenever ANY prop or input changes.
+  /**
+   * Handle the user pressing the Export button and starting a recording.
+   */
+  const handleStartExport = (): void => {
+    if (isRecording) {
+      return;
+    }
+    setIsRecording(true);
+
+    /** Min and max are both inclusive */
+    let min: number, max: number;
+    switch (exportMode) {
+      case ExportMode.ALL:
+        min = 0;
+        max = props.totalFrames - 1;
+        break;
+      case ExportMode.CURRENT:
+        min = props.currentFrame;
+        max = props.currentFrame;
+        break;
+      case ExportMode.CUSTOM:
+        // Clamp range values in case of unsafe input
+        min = clamp(customMin, 0, props.totalFrames - 1);
+        max = clamp(customMax, min, props.totalFrames - 1);
+    }
+
+    // Start the recording
+    props.startRecording({
+      min: min,
+      max: max,
+      prefix: imagePrefix,
+      minDigits: (props.totalFrames - 1).toString().length,
+      frameIncrement: frameIncrement,
+      onCompletedCallback: () => {
+        // Close modal once recording finishes and show completion notification
+        setPercentComplete(100);
+        notification.success({
+          message: "Export complete.",
+          placement: "bottomLeft",
+          duration: 4,
+          icon: <CheckCircleOutlined style={{ color: theme.color.text.success }} />,
+        });
+        // Close the modal after a small delay so the success notification can be seen
+        setTimeout(() => stopRecording(true), 750);
```

Comment:
Nit: this kind of thing always scares me.  It makes me wonder how our app would respond to "crazy clicking" where you just go around clicking on everything randomly, really fast and see what breaks.

Reply:
Valid! I added an extra flag for when the closing animation is running because it was possible to stop and restart the recording (and then have the modal close on you :X).

It is _also_ possible to have the Cancel confirmation stay open after the window has been closed, but I don't think this causes any aberrant behavior (clicking either option won't reopen the Export window) so it's probably fine.